cmd/kubeadm/app/cmd/join.go

```diff
@@ -214,6 +248,12 @@ func NewJoin(cfgPath string, args []string, cfg *kubeadmapi.NodeConfiguration, i
 		}
 	}
 
+	// Enforce usage of --master flag with --feature-gates=HighAvailability
+	if (cfg.Master == true) &&
+		!features.Enabled(cfg.FeatureGates, features.HighAvailability) {
```

Comment:
That is some weird UX on the command line.

Reply:
Being an alpha feature, IMO the use should explicitly opt-in even if the UX is kind of weird.
However no problem to change this if there is agreement on a different approach.